source/val/validate_annotation.cpp

```diff
@@ -269,6 +287,14 @@ spv_result_t ValidateMemberDecorate(ValidationState_t& _,
            << "' is not valid for the WebGPU execution environment.";
   }
 
+  if (spvIsVulkanEnv(_.context()->target_env) &&
+      !IsValidVulkanDecoration(decoration)) {
```

Comment:
The decoration only be applied to a structure type, so you needn't check for it here. It would be better to have an overall check elsewhere for correct member decorations.

Reply:
I see what you mean, I can make a followup PR that goes through the spec and for the 

> Apply only to a structure type member

type of decorations and add a check in `ValidateMemberDecorate`